### ISSUE TITLE
`require-selections` does not report issues on `FragmentSpread`s within `GraphQLUnionType`

### ISSUE THREAD

**Issue by maciesielka** [dimaMachina/graphql-eslint#2318]
### Issue workflow progress

_Progress of the issue based on the
[Contributor Workflow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow)_

- [x] 1. The issue provides a reproduction available on GitHub, Stackblitz or CodeSandbox
- [x] 2. A failing test has been provided
- [x] 3. A local solution has been provided
- [x] 4. A pull request is pending review

---

**Describe the bug**

`require-selections` handles only `InlineFragment` spreads with `GraphQLUnionType`s, despite named `FragmentSpread`s being valid GraphQL that should receive the same treatment.

https://github.com/dimaMachina/graphql-eslint/blob/58467e65f0283f5ad7a7ca8a3f0301b7669b4416/packages/plugin/src/rules/require-selections.ts#L172-L182

**To Reproduce** 

The failing test in the linked PR provides the best example for the bug (see [commit](https://github.com/dimaMachina/graphql-eslint/pull/2319/commits/835a02f99124a9190e6baa83ec091b5bb56aea49) that adds that test), but a simplified schema is reproduced below:
```graphql
type User {
  id: ID
  name: String
  posts: [Post]
}

type Post {
  id: ID
  title: String
  author: [User!]!
}

type Query {
  user: User
  userOrPost: UserOrPost
}

union UserOrPost = User | Post

fragment UnionFragment on UserOrPost {
  ...UserFields
}

fragment UserFields on User {
  name
}

query {
  userOrPost {
    # no error is reported here, even though `...UnionFragment` does not select `id`
    ...UnionFragment
  }
}
```

**Expected behavior**

An error should be reported for `FragmentSpread` just like it is for `InlineFragment`s within `GraphQLUnionType`s.

**Environment:**

- `@graphql-eslint/eslint-plugin`: 4.0.0
